### PR TITLE
Event Register: Map pop up

### PR DESCRIPTION
**Pull request by sergiomonz**
Getting coordinates from pop up map in event register Form.

### CODE REVIEW

**Comment by sergiomonz** [comment]
image
image
image

File: dashboard/src/components/Events/Register.vue [hunk 1]
```diff
@@ -157,36 +205,131 @@ export default {
...
-      { category: "Musica" },
-      { category: "Educacion" },
-    ],
+    openDialog() {
+      if (this.showModal) {
+        this.showModal=!this.showModal
+      }
+    },
+    coordenadas: function addMarker(event) {
+      //this.markers.push(event.latlng);
```
Comment (1701722767) [inline dashboard/src/components/Events/Register.vue]: delete this

File: dashboard/src/components/Events/Register.vue [hunk 2]
```diff
@@ -139,54 +230,242 @@
...
+  },
+  mounted() {
+    this.getCategories();
+  },
   methods: {
-    validate() {
-      if (this.$refs.loginForm.validate()) {
-        // submit form to server/API here...
+    register() {
+      if (!this.$refs.registerForm.validate()) {
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue:258]: show notification to user

File: dashboard/src/components/Events/Register.vue [hunk 3]
```diff
@@ -139,54 +230,242 @@
...
+      postJSON("/events", this.model, true)
+        .then((res) => {
+          this.loading = false;
+
+          if(res.error){
+            notification({
+            message: res.message,
+            icon: "mdi-alert-circle"
+          });
+          }else{
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: delete else and add return in if

File: dashboard/src/components/Events/Register.vue [hunk 4]
```diff
@@ -139,54 +230,242 @@
...
+            notification({
+            message: res.message,
+            icon: "mdi-check-bold"
+          });
+          } 
+
+          if (!res.error) {
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: in then delete this if

File: dashboard/src/components/Events/Register.vue [hunk 5]
```diff
@@ -139,54 +230,242 @@
...
+          notification({
+            message: "Ocurrió un error al hacer la petición",
+          });
+        });
     },
+    getCategories (){
+        getJSON("/categories", this.params, false)
+          .then((res) => {
+            this.categories = res.data.items;
+            console.log(res);
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: detele this

File: dashboard/src/components/Events/Register.vue [hunk 6]
```diff
@@ -139,54 +230,242 @@
...
+          })
+          .catch((err) => {
+            console.log(err);
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: inform error to user

File: dashboard/src/components/Events/Register.vue [hunk 7]
```diff
@@ -139,54 +230,242 @@
...
-
-    show1: false,
-    rules: {
-      required: (value) => !!value || "Campo requerido",
-      min: (v) => (v && v.length >= 8) || "Debe contener al menos 8 caracteres",
+    addMarker(event) {
+      this.markers.push(event.latlng);
+      let latitude= event.latlng.lat;
+      let longitude= event.latlng.lng;
+      console.log(event);
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: delete this

File: dashboard/src/components/Events/Register.vue [hunk 8]
```diff
@@ -139,54 +230,242 @@
...
-      { category: "Musica" },
-      { category: "Educacion" },
-    ],
-
-    show1: false,
-    rules: {
-      required: (value) => !!value || "Campo requerido",
-      min: (v) => (v && v.length >= 8) || "Debe contener al menos 8 caracteres",
+    addMarker(event) {
+      this.markers.push(event.latlng);
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: is it for?

File: dashboard/src/components/Events/Register.vue [hunk 9]
```diff
@@ -139,54 +230,242 @@
...
+    return {
+      url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+      attribution:
+          '&copy; <a target="_blank" href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+      zoom: 16,
+      center: [5.05690, -75.50356],
+       markers:[
+        L.latLng(5.05690, -75.50356),
+      ],
+      menu: false,
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: menu of what?

File: dashboard/src/components/Events/Register.vue [hunk 10]
```diff
@@ -139,54 +230,242 @@
...
+        max:"",
+        range_age:[0,1],
+        //images: null,
+        datestart: "",
+        dateend: "",
+        category_id:"",
+        coordinates:{}
+      },
+      date1:null,
+      date2:null,
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: improve var name

File: dashboard/src/components/Events/Register.vue [hunk 11]
```diff
@@ -139,54 +230,242 @@
...
+      categories:"",
+      select: { category: "" },
+      items: [
+        { category: "Fiesta" },
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: delete this mocks

File: dashboard/src/components/Events/Register.vue [hunk 12]
```diff
@@ -139,54 +230,242 @@
...
+        'mdi-human-male',
+        'mdi-human-male',
+        'mdi-human-male',
+        'mdi-human-male',
+      ],
+      show1: false,
+      showModal:false,
+      rules: {
+        required: (value) => !!value || "Campo requerido",
+        min: (v) => (v && v.length >= 8) || "Debe contener al menos 8 caracteres",
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: is it necessary?

File: dashboard/src/components/Events/Register.vue [hunk 13]
```diff
@@ -139,54 +230,242 @@
...
+      }
+    };
+  }
 };
 </script>
 
-<style></style>
+<style>
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: add scoped in this styles for not break other styles in the app

File: dashboard/src/components/Map.vue [hunk 14]
```diff
@@ -39,3 +39,9 @@ export default {
   },
 };
 </script>
+
+<style>
+  .leaflet-marker-icon {
```
Comment (batoJC) [inline dashboard/src/components/Map.vue]: delete this

File: dashboard/src/views/myEvents.vue [hunk 15]
```diff
@@ -0,0 +1,31 @@
+<template>
+    <div class="content">
+        <router-view></router-view>
+        <v-col cols="12" sm="6" md="6">
+                      <v-btn
```
Comment (batoJC) [inline dashboard/src/views/myEvents.vue]: fix format in this file

File: dashboard/src/views/myEvents.vue [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+                        :to="{ path: '/event-register'}"
+                        >Crear evento
+                        </v-btn>
+        </v-col>
+    </div>
+</template>
+
+<script>
+export default {
+    name:"myEvents",
```
Comment (batoJC) [inline dashboard/src/views/myEvents.vue]: MyEvents*

File: dashboard/src/views/myEvents.vue [hunk 17]
```diff
@@ -0,0 +1,31 @@
+<template>
+    <div class="content">
+        <router-view></router-view>
```
Comment (batoJC) [inline dashboard/src/views/myEvents.vue]: delete this

File: dashboard/src/views/myEvents.vue (hunk 16 above)
Comment (batoJC) [reply dashboard/src/views/myEvents.vue]: and file name

**Review by batoJC — CHANGES_REQUESTED** [review]

File: dashboard/src/components/Events/Register.vue (hunk 5 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: which one?

File: dashboard/src/components/Map.vue (hunk 14 above)
Comment (sergiomonz) [reply dashboard/src/components/Map.vue]: Done

File: dashboard/src/components/Events/Register.vue (hunk 8 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: For getting coordinates by clicking the screen.
Min in rules: Deleted

Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: Marker pushing: Deleted

File: dashboard/src/components/Events/Register.vue (hunk 10 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: Not used anymore. Deleted

File: dashboard/src/components/Events/Register.vue (hunk 11 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: deleted

File: dashboard/src/components/Events/Register.vue (hunk 12 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: deleted

File: dashboard/src/components/Events/Register.vue (hunk 9 above)
Comment (sergiomonz) [reply dashboard/src/components/Events/Register.vue]: Not longer used

File: dashboard/src/components/Events/Register.vue [hunk 18]
```diff
@@ -139,54 +230,229 @@
...
     },
+    getCategories (){
+        getJSON("/categories", this.params, false)
+          .then((res) => {
+            this.categories = res.data.items;
+          })
+          .catch((err) => {
+            console.log(err);
+            notification({
+            message: err.message,
```
Comment (batoJC) [inline dashboard/src/components/Events/Register.vue]: err.message not exist in this context

Comment (batoJC) [reply dashboard/src/components/Events/Register.vue]: only show "Error al obtener las categorias"

**Review by batoJC — CHANGES_REQUESTED** [review]

**Review by 1701722767 — APPROVED** [review]

**Review by nicolascr181 — APPROVED** [review]

**Pull request merged**